qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -352,19 +319,58 @@ def setup_default_layouts(self, window_settings):
                     row[0].dockwidget.show()
                     row[0].dockwidget.raise_()
 
-    def save_current_window_settings(self, prefix, section='main'):
-        """Save current window settings with *prefix* in
-        the userconfig-based configuration, under *section*"""
-        win_size = self.window_size
-
-        CONF.set(section, prefix + 'size', (win_size.width(), win_size.height()))
-        CONF.set(section, prefix + 'is_maximized', self.isMaximized())
-        CONF.set(section, prefix + 'is_fullscreen', self.isFullScreen())
-        pos = self.window_position
-        CONF.set(section, prefix + 'position', (pos.x(), pos.y()))
-        self.maximize_dockwidget(restore=True)  # Restore non-maximized layout
-        qba = self.saveState()
-        CONF.set(section, prefix + 'state', qbytearray_to_str(qba))
+            # set the width and height
+            self._layout_widget_info = []
```

Comment:
As discussed offline. This doesn't work properly for X11 (for some reason) so I think it's best to revisit it after the initial demo as without the layout is nearly what we want anyway.